### PR TITLE
feature(subgraph): Adding missing properties

### PR DESCRIPTION
**Pull request by julien51**
# Description

In order to finally ship the "state of Unlock" page, I am adding 2 missing properties to our subgraphs. This PR also fixes a bug when initializing UnlockDailyData objects.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: subgraph/src/helpers.ts [hunk 1]
```diff
@@ -34,3 +35,29 @@ export function getKeyExpirationTimestampFor(
     return lockContract.keyExpirationTimestampFor(ownerAddress)
   }
 }
+
+export function loadOrCreateUnlockDailyData(timestamp: number) {
+  const dayID = timestamp / 86400
+  let unlockDailyData = UnlockDailyData.load(dayID.toString())
+  const unlockDailyDataYesterday = UnlockDailyData.load((dayID - 1).toString())
+  if (unlockDailyData === null) {
```
Comment (clemsos) [inline subgraph/src/helpers.ts:43]: doesnt that condition mean that no action will be taken if yesterday data is not null?

File: subgraph/schema.graphql [hunk 2]
```diff
@@ -53,15 +53,19 @@ type Key @entity {
...
   lockDeployed: BigInt!
-  "Daily number of keys sold" 
+  "Total number of locks deployed"
+  totalLockDeployed: BigInt!
+  "Daily number of keys sold"
   keysSold: BigInt!
-  "Daily number of active locks" 
+  "Total number of keys sold"
+  totalKeysSold: BigInt!
+  "Daily number of active locks"
```
Comment (clemsos) [inline subgraph/schema.graphql:66]: how is "active" defined?

File: subgraph/src/public-lock.ts [hunk 3]
```diff
@@ -50,17 +55,18 @@ function newKey(event: TransferEvent): void {
...
-  if (unlockDailyData) {
-    const activeLocks = unlockDailyData.activeLocks
-    unlockDailyData.keysSold = unlockDailyData.keysSold.plus(BigInt.fromI32(1))
-    if (activeLocks && !activeLocks.includes(event.address)) {
-      activeLocks.push(event.address)
-      unlockDailyData.activeLocks = activeLocks
-    }
-    unlockDailyData.save()
+  const timestamp = event.block.timestamp.toI32()
+  const unlockDailyData = loadOrCreateUnlockDailyData(timestamp)
```
Comment (clemsos) [inline subgraph/src/public-lock.ts:59]: does `event.block.timestamp.toI32()` directly gives a day? should be in seconds so there is one missing step here

File: subgraph/src/unlock.ts [hunk 4]
```diff
@@ -31,23 +31,16 @@ export function handleNewLock(event: NewLock): void {
...
-    unlockDailyData.save()
-  } else {
-    unlockDailyData.lockDeployed = unlockDailyData.lockDeployed.plus(
-      BigInt.fromI32(1)
-    )
-    unlockDailyData.save()
-  }
+  // update lockDayData
+  const timestamp = event.block.timestamp.toI32()
+  const unlockDailyData = loadOrCreateUnlockDailyData(timestamp)
```
Comment (clemsos) [inline subgraph/src/unlock.ts:36]: same here

File: subgraph/src/unlock.ts [hunk 5]
```diff
@@ -122,10 +114,7 @@ export function handleLockUpgraded(event: LockUpgraded): void {
 
 export function handleGNPChanged(event: GNPChanged): void {
   let timestamp = event.block.timestamp.toI32()
-  let dayID = timestamp / 86400
-  let unlockDailyData = UnlockDailyData.load(dayID.toString())
-  if (unlockDailyData) {
-    unlockDailyData.grossNetworkProduct = event.params.grossNetworkProduct
-    unlockDailyData.save()
-  }
+  const unlockDailyData = loadOrCreateUnlockDailyData(timestamp)
```
Comment (clemsos) [inline subgraph/src/unlock.ts:117]: same here

File: subgraph/src/public-lock.ts (hunk 3 above)
Comment (julien51) [reply subgraph/src/public-lock.ts:59]: We compute the day from the timestamp in `loadOrCreateUnlockDailyData`: `  const dayID = timestamp / 86400
`

**Comment by searchableguy** [comment]
Question: Do we want to include the new objects added for state of the unlock into subgraph service as well?

**Review by searchableguy — APPROVED** [review]

**Comment by julien51** [comment]
> Question: Do we want to include the new objects added for state of the unlock into subgraph service as well?


That's a good idea! I actually don't know how to do that?

**Comment by searchableguy** [comment]
@julien51 

https://github.com/unlock-protocol/unlock/blob/master/packages/unlock-js/src/subgraph/schema.graphql

You will need to add a query with variables that you want others to be able to pass or filter in graphql.

This will automatically generates types and sdk.

Then you can export it as it is or wrap it to make it work for multiple networks at once in https://github.com/unlock-protocol/unlock/blob/master/packages/unlock-js/src/subgraph/index.ts

**Comment by julien51** [comment]
Perfect! Let me add that to this PR!

**Comment by julien51** [comment]
ok so I just added the `deployer`. I don't think the data fields are useful for unlockjs... they are mostly analytics.

**Pull request merged**